lldb/test/API/commands/statistics/basic/main.cpp

```diff
@@ -1,7 +1,13 @@
 // Test that the lldb command `statistics` works.
+#include <string>
+
+void foo() {
+  std::string str = "hello world";
```

Comment:
We should also test a type that uses a template as some summary providers match based on a regex like:
```
std::vector<int> ints = {1, 2, 3};
std::vector<float> floats = {1.0, 2.0, 3.0};
```
From debugging we can see:
```
(lldb) type summary info ints
summary applied to (std::vector<int>) ints is:  (show children) (hide value) libc++ std::vector summary provider
```
And the summary definition looks like:
```
(lldb) type summary list
...
^std::vector<.+>(( )?&)?$: `size=${svar%#}` (show children) (hide value)
...
```

So the summary name we want to see in our resulting map is probably `"^std::vector<.+>(( )?&)?$"`

Reply:
@clayborg 

I created a basic template test, and made sure the formatter was invoked, and I added a different test for the two vector cases.

I did not test nested summaries, and will add it if desired.